pkg/instrumentation/sdk.go

```diff
@@ -51,107 +51,154 @@ type sdkInjector struct {
+
+		for _, container := range strings.Split(pythonContainers, ",") {
+			index := getContainerIndex(container, pod)
+			pod, err = injectPythonSDK(otelinst.Spec.Python, pod, index)
+			if err != nil {
+				i.logger.Info("Skipping Python SDK injection", "reason", err.Error(), "container", pod.Spec.Containers[index].Name)
+			} else {
+				pod = i.injectCommonEnvVar(otelinst, pod, index)
+				pod = i.injectCommonSDKConfig(ctx, otelinst, ns, pod, index, index)
+			}
 		}
 	}
-	if insts.DotNet != nil {
-		otelinst := *insts.DotNet
+	if insts.DotNet.Instrumentation != nil {
+		otelinst := *insts.DotNet.Instrumentation
 		var err error
 		i.logger.V(1).Info("injecting DotNet instrumentation into pod", "otelinst-namespace", otelinst.Namespace, "otelinst-name", otelinst.Name)
-		pod, err = injectDotNetSDK(otelinst.Spec.DotNet, pod, index)
-		if err != nil {
-			i.logger.Info("Skipping DotNet SDK injection", "reason", err.Error(), "container", pod.Spec.Containers[index].Name)
-		} else {
-			pod = i.injectCommonEnvVar(otelinst, pod, index)
-			pod = i.injectCommonSDKConfig(ctx, otelinst, ns, pod, index, index)
+
+		dotnetContainers := insts.DotNet.Containers
+
+		for _, container := range strings.Split(dotnetContainers, ",") {
+			index := getContainerIndex(container, pod)
+			pod, err = injectDotNetSDK(otelinst.Spec.DotNet, pod, index)
+			if err != nil {
+				i.logger.Info("Skipping DotNet SDK injection", "reason", err.Error(), "container", pod.Spec.Containers[index].Name)
+			} else {
+				pod = i.injectCommonEnvVar(otelinst, pod, index)
+				pod = i.injectCommonSDKConfig(ctx, otelinst, ns, pod, index, index)
+			}
 		}
 	}
-	if insts.Go != nil {
+	if insts.Go.Instrumentation != nil {
 		origPod := pod
-		otelinst := *insts.Go
+		otelinst := *insts.Go.Instrumentation
 		var err error
 		i.logger.V(1).Info("injecting Go instrumentation into pod", "otelinst-namespace", otelinst.Namespace, "otelinst-name", otelinst.Name)
-		pod, err = injectGoSDK(otelinst.Spec.Go, pod)
-		if err != nil {
-			i.logger.Info("Skipping Go SDK injection", "reason", err.Error(), "container", pod.Spec.Containers[index].Name)
-		} else {
-			// Common env vars and config need to be applied to the agent contain.
-			pod = i.injectCommonEnvVar(otelinst, pod, len(pod.Spec.Containers)-1)
-			pod = i.injectCommonSDKConfig(ctx, otelinst, ns, pod, len(pod.Spec.Containers)-1, 0)
-
-			// Ensure that after all the env var coalescing we have a value for OTEL_GO_AUTO_TARGET_EXE
-			idx := getIndexOfEnv(pod.Spec.Containers[len(pod.Spec.Containers)-1].Env, envOtelTargetExe)
-			if idx == -1 {
-				i.logger.Info("Skipping Go SDK injection", "reason", "OTEL_GO_AUTO_TARGET_EXE not set", "container", pod.Spec.Containers[index].Name)
-				pod = origPod
+
+		goContainers := insts.Go.Containers
+
+		// Go instrumentation supports only single container instrumentation. Loop here should be replaced/improved.
+		for _, container := range strings.Split(goContainers, ",") {
```

Comment:
Can this section work off the assumption that `goContainer` is exactly 1 container? We could take the value as is.

Reply:
Yeah, you're right :)